runtime/commonMain/src/kotlinx/benchmark/CommonBlackhole.kt

```diff
@@ -0,0 +1,42 @@
+package kotlinx.benchmark
+
+private const val MAGIC_SIZE: Int = 13
+
+open class CommonBlackhole {
+    private val arrayOfAny: Array<Any?> = arrayOfNulls(MAGIC_SIZE)
+    private var currentAnyPosition: Int = 0
+    private fun consumeAny(obj: Any?) {
+        arrayOfAny[currentAnyPosition] = obj
+        currentAnyPosition = if (currentAnyPosition == MAGIC_SIZE - 1) 0 else currentAnyPosition + 1
+    }
+
+    private val arrayOfInt: IntArray = IntArray(MAGIC_SIZE)
+    private var currentIntPosition: Int = 0
+    private fun consumeInt(i: Int) {
+        arrayOfInt[currentIntPosition] = i
+        currentIntPosition = if (currentIntPosition == MAGIC_SIZE - 1) 0 else currentIntPosition + 1
+    }
+
+    fun flushMe() {
+        val sums = arrayOfAny.sumOf { it?.hashCode() ?: 0 } + arrayOfInt.sum()
```

Comment:
Minor: hashCode is available on a nullable receiver, so it can be just `it.hashCode()`